contribute/design/1.x/cstor/20200629-cstor-backupandrestore-v1.md

```diff
@@ -0,0 +1,292 @@
+---
+oep-number: CStor Backup and Restore REV2
+title: Migration of CStor Backup and Restore Resources to V1
+authors:
+  - "@mittachaitu"
+  - "@sonasingh46"
+owners:
+  - "@kmova"
+  - "@vishnuitta"
+  - "@mynktl"
+editor: "@mittachaitu"
+creation-date: 2020-06-29
+last-updated: 2020-06-29
+status: provisional
+---
+
+# Migration of CStor Backup and Restore Resources to V1
+
+## Table of Contents
+
+- [Migration of CStor Backup and Restore Resources to V1](#migration-of-cstor-backup-and-restore-resources-to-v1)
+	- [Table of Contents](#table-of-contents)
+	- [Introduction](#introduction)
+	- [Motivation](#motivation)
+		- [Goals](#goals)
+		- [Non-Goals](#non-goals)
+	- [Proposal](#proposal)
+		- [Proposed Approaches](#proposed-approaches)
+		- [Steps to perform user stories](#steps-to-perform-user-stories)
+		- [Low Level Design](#low-level-design)
+			- [Work Flow](#work-flow)
+				- [Current State of Schemas](#current-state-of-schemas)
+				- [Proposed Schemas](#proposed-schemas)
+				- [Migration Workflows](#migration-workflows)
+					- [Approach1(With Backward compatibility)](#approach1with-backward-compatibility)
+					- [Cons](#cons)
+					- [Apporach2(Without Backward compatibility)](#apporach2without-backward-compatibility)
+					- [Cons](#cons-1)
+
+## Introduction
+
+CStor Backup and Restore functionality is supported by a set of OpenEBS CRs and OpenEBS Velero plugin. On a high level, the backup and restore requests are initiated by Velero and there is an OpenEBS Velero plugin that intercepts the request and does a REST call to one of the OpenEBS components with a payload. The REST server finally provisions the required CRs by processing the information passed in the payload. Once the CRs are created/deleted, the controller components of the OpenEBS control plane converge to the intent specified in the CRs by making calls to the data layer.
+
+The following are the CRs that facilitates Backup and Restore:
+- CStorBackup			-- (Reconciled by Backup controller in pool manager)
+- CStorRestore			-- (Reconciled by Restore controller in pool manger)
+- CStorCompletedBackup	-- (Used to store last completed and second last completed backup information which is used for scheduled backups)
+
+A detailed description on how cStor Backup and Restore works is out of the scope of this document. For more details to know how the cStor backup and restore works, please click [here](https://github.com/openebs/openebs/blob/master/contribute/design/1.x/cstor/20200511-cstor-backupandrestore.md)
+
+## Motivation
+
+- Create a backup and restore CRs in V1 version of cstor.openebs.io group which provides backup and restore support of cStor CSI volumes.
```

Comment:
n_P: -- > 
Promote backup and restore CRs to v1 version ....

Reply:
done